### PR TITLE
[vector_graphics*] Relax dependency constraints of vector_graphics, vector_graphics_codec, vector_graphics_compiler, flutter_svg 

### PR DESCRIPTION
**Pull request by jonahwilliams**
Relax the dependency constrraints of the vector graphics packages. Since there are no actual changes to the vector_graphics_codec package and the listed version is already published, I don't think it needs a version bump to remove the publish key.

Fixes https://github.com/flutter/flutter/issues/157626

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

### CODE REVIEW

File: packages/vector_graphics/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 1.1.14
+
+* Relax dependency constraint on vector_graphics_codec.
```
Comment (stuartmorgan) [inline packages/vector_graphics/CHANGELOG.md:3]: Relaxes

File: packages/vector_graphics/pubspec.yaml [hunk 2]
```diff
@@ -15,14 +12,12 @@ dependencies:
   flutter:
     sdk: flutter
   http: ^1.0.0
-  # See https://github.com/flutter/flutter/issues/157626
-  vector_graphics_codec: ">=1.1.11+1 <= 1.1.12"
+  vector_graphics_codec: "^1.1.11+1"
```
Comment (stuartmorgan) [inline packages/vector_graphics/pubspec.yaml:15]: These shouldn't need quotes once they are using `^`.

File: packages/vector_graphics_compiler/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 1.1.13
+
+* Relax dependency constraint on vector_graphics_codec.
```
Comment (stuartmorgan) [inline packages/vector_graphics_compiler/CHANGELOG.md:3]: Relaxes

File: third_party/packages/flutter_svg/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,8 @@
+## 2.0.12
+
+* Relax the dependency constraints on vector_graphics, vector_graphics_codec,
```
Comment (stuartmorgan) [inline third_party/packages/flutter_svg/CHANGELOG.md:3]: Relaxes

File: packages/vector_graphics_compiler/pubspec.yaml [hunk 5]
```diff
@@ -18,8 +15,7 @@ dependencies:
   meta: ^1.7.0
   path: ^1.8.0
   path_parsing: ^1.0.1
-  # See https://github.com/flutter/flutter/issues/157626
-  vector_graphics_codec: ">=1.1.11+1 <= 1.1.12"
+  vector_graphics_codec: "^1.1.11+1"
```
Comment (stuartmorgan) [inline packages/vector_graphics_compiler/pubspec.yaml:18]: No need for quotes

File: third_party/packages/flutter_svg/pubspec.yaml [hunk 6]
```diff
@@ -12,10 +12,9 @@ dependencies:
   flutter:
     sdk: flutter
   http: ^1.0.0
-  # See https://github.com/flutter/flutter/issues/157626
-  vector_graphics: ">=1.1.11+1 <= 1.1.12"
-  vector_graphics_codec: ">=1.1.11+1 <= 1.1.12"
-  vector_graphics_compiler: ">=1.1.11+1 <= 1.1.12"
+  vector_graphics: "^1.1.11+1"
+  vector_graphics_codec: "^1.1.11+1"
+  vector_graphics_compiler: "^1.1.11+1"
```
Comment (stuartmorgan) [inline third_party/packages/flutter_svg/pubspec.yaml:17]: Same for these.

**Review by stuartmorgan — APPROVED** [review]
LGTM with nits

**Comment by stuartmorgan** [comment]
Version/CHANGELOG override: Removing `publish_to: none` does not require publishing.

**Comment by stuartmorgan** [comment]
I resolved the conflicts and am marking this for autosubmit to unblock the wasm PRs.

**Comment by jonahwilliams** [comment]
Thanks @stuartmorgan !

**Pull request merged**